raiden/transfer/channel.py

```diff
@@ -242,19 +271,32 @@ def is_valid_directtransfer(
 
         result = (False, msg)
 
-    elif received_balance_proof.locked_amount != expected_locked_amount:
+    elif received_balance_proof.locked_amount != current_locked_amount:
         # Direct transfers must not change the locked_amount. Otherwise the
         # sender is trying to play the protocol and steal token.
         msg = (
             'Invalid DirectTransfer message. '
             "Balance proof's locked_amount is invalid, expected: {} got: {}."
         ).format(
-            expected_locked_amount,
+            current_locked_amount,
             received_balance_proof.locked_amount,
         )
 
         result = (False, msg)
 
+    elif is_valid_amount(sender_state, amount):
```

Comment:
Should be the reverse - invalid if the amount is not valid. Same for line 415.

Reply:
uhhhh, true, I missed that while factoring the code out